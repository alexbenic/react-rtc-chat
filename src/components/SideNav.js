import React from "react";
import PropTypes from "prop-types";
import UserList from "./UserList";
import * as Types from "../types";

import angularLogo from "../assets/images/angular_logo.png";
import reactLogo from "../assets/images/react_logo.png";

const Counter = ({ angular, react }) => {
  return (
    <div className="flex flex-column flex-row-ns flex-none-ns h4 h3-ns ma2">
      <div className="flex flex-auto flex-column justify-center items-center pa1">
        <img src={angularLogo} className="db w3 h3" alt="angular logo" />
        <span className="tc"> {angular} </span>
      </div>
      <div className="flex flex-auto flex-column justify-center items-center p1">
        <img src={reactLogo} className="db w3 h3" alt="react logo" />
        <span> {react} </span>
      </div>
    </div>
  );
};

Counter.propTypes = {
  angular: PropTypes.number.isRequired,
  react: PropTypes.number.isRequired
};

const SideNav = ({ current, users, stats }) => {
  return (
    <aside className="w-20 pa3 flex flex-column bg-sidebar">
      <img
        className="db w-100 h4 center ofc"
        src={reactLogo}
        alt="react logo"
      />
      <span title={current.nick} className="pa2 bb bw-1 tc truncate">
        {current.nick}
      </span>
      <UserList current={current} users={users} />
      <Counter angular={stats.angular} react={stats.react} />
    </aside>
  );
};

SideNav.propTypes = {
  current: Types.User.isRequired,
  users: Types.Users,
  stats: Types.Stats
};

export default SideNav;
