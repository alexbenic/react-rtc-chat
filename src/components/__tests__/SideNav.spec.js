import React from "react";
import each from "jest-each";
import { render, cleanup } from "react-testing-library";
import randombytes from "randombytes";
import SideNav from "../SideNav";

const user = name => {
  return {
    id: randombytes(8).toString("hex"),
    nick: name,
    stream: {},
    app: 1
  };
};

afterEach(cleanup);

each([
  ["zvrk", [], { angular: 0, react: 1 }],
  ["zoom", [user("zoom"), user("maci")], { angular: 1, react: 2 }],
  ["maci", [user("zoom"), user("zvrk"), user("zook")], { angular: 2, react: 2 }]
]).test(
  "current user %s, list of connected %j, and stats %o",
  (name, users, count) => {
    const { container, getByTestId } = render(
      <SideNav current={user(name)} users={users} stats={count} />
    );

    const currentName = getByTestId("current-nick");
    const userList = container.querySelectorAll("li");

    expect(currentName.textContent).toMatch(name);
    expect(userList.length).toBe(users.length);
  }
);

test("snapshot", () => {
  const { container, getByTestId } = render(
    <SideNav
      current={user("zvkr")}
      users={[]}
      stats={{ angular: 0, react: 1 }}
    />
  );

  expect(container.firstChild).toMatchSnapshot();
});
