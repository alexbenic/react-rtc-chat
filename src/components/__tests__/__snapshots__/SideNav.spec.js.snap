// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<aside
  class="w-20 pa3 flex flex-column bg-sidebar"
>
  <img
    class="db w-100 h4 center ofc"
    src="react_logo.png"
    alt="react logo"
  />
  <span
    data-testid="current-nick"
    title="zvkr"
    class="pa2 bb bw-1 tc truncate"
  >
    zvkr
  </span>
  <div
    class="flex-auto mt2"
  />
  <div
    class="flex flex-column flex-row-ns flex-none-ns h4 h3-ns ma2"
  >
    <div
      class="flex flex-auto flex-column justify-center items-center pa1"
    >
      <img
        src="angular_logo.png"
        class="db w3 h3"
        alt="angular logo"
      />
      <span
        class="tc"
      >
        0
      </span>
    </div>
    <div
      class="flex flex-auto flex-column justify-center items-center p1"
    >
      <img
        src="react_logo.png"
        class="db w3 h3"
        alt="react logo"
      />
      <span>
        1
      </span>
    </div>
  </div>
</aside>
`;
